src/modules/Game.class.js

```diff
@@ -6,63 +6,261 @@
+  static gameStatuses = {
+    idle: 'idle',
+    playing: 'playing',
+    lose: 'lose',
+    win: 'win',
+  };
+
+  constructor(
+    initialState = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ],
+  ) {
+    this.initialState = initialState;
+    this.state = this.initialState.map((row) => [...row]);
+    this.status = Game.gameStatuses.idle;
+    this.score = 0;
+  }
+
+  moveLeft() {
+    if (this.getStatus() !== Game.gameStatuses.playing) {
+      return;
+    }
+
+    const updatedState = this.state.map((row) => this.move(row));
+
+    if (JSON.stringify(this.getState()) !== JSON.stringify(updatedState)) {
+      this.updateGameState(updatedState);
+      this.addCells();
+    }
+  }
+
+  moveRight() {
+    if (this.getStatus() !== Game.gameStatuses.playing) {
+      return;
+    }
+
+    const reverseState = this.state.map((row) => [...row].reverse());
+
+    const updatedState = reverseState.map((row) => {
+      return this.move(row).reverse();
+    });
+
+    if (JSON.stringify(this.getState()) !== JSON.stringify(updatedState)) {
+      this.updateGameState(updatedState);
+      this.addCells();
+    }
+  }
+
+  moveUp() {
+    if (this.getStatus() !== Game.gameStatuses.playing) {
+      return;
+    }
+
+    const rotateState = this.rotateMatrixCounteClockwise(this.getState());
+
+    const updatedState = rotateState.map((row) => this.move(row));
+
+    const unRotateSteate = this.rotateMatrixClockwise(updatedState);
+
+    if (JSON.stringify(this.getState()) !== JSON.stringify(unRotateSteate)) {
+      this.updateGameState(unRotateSteate);
+      this.addCells();
+    }
+  }
+
+  moveDown() {
+    if (this.getStatus() !== Game.gameStatuses.playing) {
+      return;
+    }
+
+    const rotateState = this.rotateMatrixClockwise(this.getState());
+
+    const updatedState = rotateState.map((row) => this.move(row));
+
+    const unRotateSteate = this.rotateMatrixCounteClockwise(updatedState);
+
+    if (JSON.stringify(this.getState()) !== JSON.stringify(unRotateSteate)) {
+      this.updateGameState(unRotateSteate);
+      this.addCells();
+    }
+  }
+
+  move(row) {
+    const newRow = [];
+    let i = 0;
+
+    while (i < row.length) {
+      const current = row[i];
+
+      if (current) {
+        let pushed = false;
+
+        for (let j = i + 1; j < row.length; j++) {
+          const next = row[j];
+
+          if (next === current) {
+            newRow.push(current * 2);
+            this.updateGameScore(current * 2);
+            pushed = true;
+            i = j + 1;
+            break;
+          } else if (next) {
+            newRow.push(current);
+            pushed = true;
+            i = j;
+            break;
+          }
+        }
+
+        if (!pushed) {
+          newRow.push(current);
+          i++;
+        }
+      } else {
+        i++;
+      }
+    }
+
+    while (newRow.length < row.length) {
+      newRow.push(0);
+    }
+
+    return newRow;
```

Comment:
The 'move' function performs the logic for moving and merging tiles in one direction. Ensure that the logic correctly handles the merging of tiles according to the game rules and that the function is tested thoroughly.